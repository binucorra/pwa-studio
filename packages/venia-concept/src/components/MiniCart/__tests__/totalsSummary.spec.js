import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import TotalsSummary from '../totalsSummary';

test('renders correctly when it has a subtotal', () => {
    const props = {
        cart: {
            details: {
                currency: {
                    quote_currency_code: 'US'
                },
                items_qty: 1
            },
            totals: {
                subtotal: 99
            }
        }
    };

    const tree = createTestInstance(<TotalsSummary {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders an empty div when it does not have a subtotal', () => {
    const props = {
        cart: {
            details: {
                currency: {
                    quote_currency_code: 'US'
                },
                items_qty: 1
            },
            totals: {}
        }
    };

    const tree = createTestInstance(<TotalsSummary {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
