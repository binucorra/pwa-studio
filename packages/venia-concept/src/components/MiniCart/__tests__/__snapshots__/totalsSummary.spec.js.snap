// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an empty div when it does not have a subtotal 1`] = `<div />`;

exports[`renders correctly when it has a subtotal 1`] = `
<div>
  <dl>
    <dt>
      <span>
        Cart Total :Â 
        <div />
      </span>
    </dt>
    <dd>
      (
      1
       
      item
      )
    </dd>
  </dl>
</div>
`;
